src/tools/rbd_mirror/image_replayer/CreateImageRequest.cc

```diff
@@ -275,6 +307,46 @@ void CreateImageRequest<I>::clone_image() {
     close_remote_parent_image();
     return;
   }
+  if (m_mirror_image_mode == cls::rbd::MIRROR_IMAGE_MODE_SNAPSHOT) {
```

Comment:
I would prefer if you did this check in `handle_open_local_parent_image`. This way it would be more clear why we needed to open the local parent image. 

This would also mean that the checks above should be moved to `handle_open_remote_parent_image` and member variables `m_snap_name` and  `m_snap_namespace` introduced to store the result.

Ideally, I think I would like we introduce `image_replayer::ValidateSnapshot` request (or something like this) that would call `image_replayer::snapshot::ValidateSnapshot` for the snapshot based mirroring, where this code could be moved. And for the journal based mirroring, it could be noop, or it could also have ``image_replayer::journal::ValidateSnapshot` that would just check the image snapshot exists. But I will not insist on this.

Reply:
@idryomov and I did discuss having a separate snapshot/journal class implementations but thought this approach would be ok. 

I'll look into moving the checks into the open functions.